src/third_party/geospatial/geospatial_aggr.cpp

```diff
@@ -43,8 +174,180 @@ TStayPointDetector::TStayPointDetector(const TWPt<TBase>& Base, const PJsonVal&
     LocationFieldId = Store->GetFieldId(LocationFieldName);
     TStr AccuracyFieldName = ParamVal->GetObjStr("accuracyField");
     AccuracyFieldId = Store->GetFieldId(AccuracyFieldName);
+
+    //init the state in case saveJson is called before onAdd();
+    State = TJsonVal::NewArr();
+}//TStayPointDetector::constructor
+
+void TStayPointDetector::OnAddRec(const TRec& Rec,
+    const TWPt<TStreamAggr>& CallerAggr)
+{
+    TScopeStopWatch StopWatch(ExeTm);//auto stopwatch for the QM aggregate
+                                        //parse new GPS record from JSON
+
+    int NumOfRecordsToForget = 0;
+    hasFinishedGeoActs = false;
+    //delete previously detected stuff
+    for (int i = 0; i < DetectedGeoActivities.Len(); i++) {
```

Comment:
Typically we use suffix V to indicate vectors, e.g. `DetectedGeoActivities` => `DetectedGeoActivityV`